### PR TITLE
feat: add possibility to pass a config object directly and add tests

### PR DESCRIPTION
**Pull request by dword-design**
The current implementation uses configPath to load the config, this PR allows to directly pass the config to the loader.

### CODE REVIEW

**Comment by dword-design** [comment]
@tom76kimo I had a look at this again and I think it needs some talking. I basically changed the (multiple) config paths to (multiple) configs, which I think does not entirely make sense, because I guess there is no use case of passing multiple config objects :P. What are the use cases for multiple config paths in the first place? Let's clarify use cases for both config object and config paths and then I could adjust it, and maybe make it non-breaking.

**Comment by redonkulus** [comment]
@dword-design I'm not sure on the original intent for multiple but I wouldn't remove it unless we do not want that feature. I would prefer to keep current functionality but then add on your enhancement as a preference option (falling back to current logic).

**Comment by dword-design** [comment]
@redonkulus I'd say merge the rules PR first and then continue with this one. We could make it analogous by requiring if it's a string and using directly if it's an object. And keep the config array for now.

**Comment by redonkulus** [comment]
@dword-design you want to clean up the conflicts and continue on with this now?

**Comment by dword-design** [comment]
@redonkulus Alright should be merged now. I was wrong, the current implementation should be non-breaking. It adds a `config` option but still has the `configPaths` option. As said, I think the config and configPath thing needs some work because currently it's a bit messy how to pass configs, but yeah you can directly pass a config with this PR ^^.

File: tests/atomicLoader.js [hunk 1]
```diff
@@ -38,4 +38,51 @@ describe('atomic loader', () => {
...
+
+    it('config', (done) => {
+        this.async = () => (err, source) => {
+            const cssFile = fs.readFileSync('./build/css/atomic.css', 'utf8');
+            expect(/\.Bgc\\\(foo\\\)/.test(cssFile)).to.equal(true);
+            done();
+        };
+        this.query = {
+            config: {
+                configs: {
```
Comment (redonkulus) [inline tests/atomicLoader.js:78]: Why `configs` under the `config` property?

File: README.md [hunk 2]
```diff
@@ -30,11 +30,11 @@ yarn add -D webpack-atomizer-loader
 
 The loader accepts the below options:
 
-| Option | Default | Required | Description |
-| ------ | ------- | -------- | ----------- |
+| Option           | Default     | Required | Description                                                                                                                                                               |
```
Comment (redonkulus) [inline README.md:33]: Looks like this change got reverted

File: lib/atomicLoader.ts [hunk 3]
```diff
@@ -102,18 +94,23 @@ const atomicLoader = function (source, map) {
...
-    let configPaths = query.configPath;
-    if (!Array.isArray(configPaths)) {
-        configPaths = [configPaths];
+    let configs = [
+        ...[].concat(configPath).map((configPath) => require(require.resolve(configPath))),
+        ...(config !== undefined ? [config] : []),
+    ];
+
+    if (configs.length === 0) {
+        configs.push(configObject.default);
```
Comment (redonkulus) [inline lib/atomicLoader.ts:109]: Is `.default` always defined? Before it checked without it first then, fellback to checking for it.

File: tests/atomicLoader.js (hunk 1 above)
Comment (dword-design) [reply tests/atomicLoader.js:78]: I made it analogous to `configPath`. Inside the config there is `configs` again.

File: lib/atomicLoader.ts (hunk 3 above)
Comment (dword-design) [reply lib/atomicLoader.ts:109]: https://github.com/acss-io/webpack-atomizer-loader/blob/master/lib/atomicLoader.ts#L23 seems to be constant.

File: README.md (hunk 2 above)
Comment (dword-design) [reply README.md:33]: hmm is it only the formatting? I ran `npm run format` again an that's the current result.

Comment (redonkulus) [reply README.md:33]: Good point, prettier seems to mess up markdown tables.

**Review by redonkulus — APPROVED** [review]
LGTM. thanks!

**Comment by redonkulus** [comment]
Released in https://github.com/acss-io/webpack-atomizer-loader/releases/tag/v1.3.0

**Comment by dword-design** [comment]
Great, thanks for merging 🙌.

**Pull request merged**